File: src/Api/Model/DB/CNAE/CnaeClasse.cs
using Newtonsoft.Json;
using Realms;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CriadorBaseDados.Model.DB.CNAE
{
    /// <summary>
    /// Seção -> Divisão -> Grupo -> Classe -> Subclasse -> Atividade Econômica
    /// </summary>
    [JsonObject(MemberSerialization.OptIn)]
    public class CnaeClasse : RealmObject
    {
        [JsonProperty]
        [PrimaryKey]
        public int ID { get; set; }
        [JsonProperty]
        public string Descricao { get; set; }
        [Ignored]
        [JsonProperty]
        public int CodGrupo => Grupo.ID;
        public CnaeGrupo Grupo { get; set; }
        [Backlink(nameof(CnaeSubclasse.Classe))]
        //[JsonProperty]
        public IQueryable<CnaeSubclasse> Subclasses { get; }
    }
}

File: src/Mobile/b2b/lib/model/core/cnae/cnae_classe.dart
import 'cnae_subclasse.dart';

class CnaeClasse {
  final int id;
  final String descricao;
  final int codGrupo;
  final List<CnaeSubclasse> subclasses;

  CnaeClasse(this.id, this.descricao, this.codGrupo, this.subclasses);

  CnaeClasse.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        descricao = json['descricao'],
        codGrupo = json['codGrupo'],
        subclasses = json['subClasses'] ?? List.empty(growable: true);

  Map<String, dynamic> toMap() => <String, dynamic>{
        'id': id,
        'descricao': descricao,
        'codGrupo': codGrupo,
        //'subClasses': subClasses,
      };
}
